hack/update-generated-files

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+# Forked from: https://github.com/moby/buildkit/blob/v0.10.4/hack/update-generated-files
+# Copyright The BuildKit Authors.
+# Copyright The Buildx Authors.
+# Licensed under the Apache License, Version 2.0
```

Comment:
I think it's fine for now but in a follow-up, we could just build using the `hack/dockerfiles/generated-files.Dockerfile` from BuildKit repo that would be fetched using the BuildKit reference from `go.mod` so we would not need an extra dockerfile.

I will also take a look at cherry picking https://github.com/moby/buildkit/pull/3204/commits/3fd07850e6eb3636472ca12057319c8736399666 so we use the go tools build constraint and avoid hardcoded versions in the Dockerfile and fully relies on ones from `go.mod` for protobuf/gogo tooling.

Reply:
https://github.com/moby/buildkit/pull/3236 is merged now. Should be easier with https://github.com/moby/buildkit/blob/master/hack/update-generated-files

As we are using bake in buildx repo it would look like this:
* https://github.com/crazy-max/buildkit/blob/44172e1b25d3f09be063c3dc277d68cdcc709d15/docker-bake.hcl#L65-L70